### ISSUE TITLE
Support for MSSQL

### ISSUE THREAD

**Issue by tomaszglinski** [kagkarlsson/db-scheduler#21]
I used the table definition below and it looks like it is working, about I cannot be really sure :)

```
create table scheduled_tasks (
  task_name varchar(40) not null,
  task_instance varchar(200) not null,
  task_data varchar(max),
  execution_time datetime2 not null,
  picked BIT not null,
  picked_by varchar(50),
  last_success datetime2 null,
  last_failure datetime2 null,
  last_heartbeat datetime2 null,
  version BIGINT not null,
  PRIMARY KEY (task_name, task_instance)
)
```

**Comment by kagkarlsson** [comment]
If you would help me test that it works, I can add it :)

Things to keep in mind: 
* timestamp should preferably contain zone information, i.e. be an instant in time. Would `datetimeoffset` be better than `datetime2`?  
* `task_data` will hold binary data, does it work with the `varchar` type in mssql?

**Comment by tomaszglinski** [comment]
Unfortunately im not an expert with MSSQL, im forced to use it:) Your remarks looks well, I will try that too . 
My data is small and worked
I didn't have issues with dates, yet it is probably so to handle clients with different zone settings? Would'n it be more efficient to store unified format like UTC?

**Comment by tomaszglinski** [comment]
Tell me how i can help, i will try. Im using https://hub.docker.com/r/microsoft/mssql-server-linux/ version

**Comment by kagkarlsson** [comment]
One thing you could do, if you want to be more sure of the compatibility is to run a _compatibility_ test for MSSQL. For example, modify [OracleCompatibilityTest](https://github.com/kagkarlsson/db-scheduler/blob/master/src/test/java/com/github/kagkarlsson/scheduler/compatibility/OracleCompatibilityTest.java) and run it against MSSQL (will require a jdbc-driver in pom.xml).

I would like to create a test-suite that is run against a set of docker containers, representing the common databases, but haven't been prioritized.

And regarding time zone, I think the efficiency gain would be negligible. It is much easier to just keep full _instants_ in the db.

**Comment by tomaszglinski** [comment]
OK, will try. Btw, I tried to use datetimeoffset for dates, but MSSQL
silently set the type to VARCHAR...

2018-03-22 7:53 GMT+01:00 Gustav Karlsson <notifications@github.com>:

**Comment by kagkarlsson** [comment]
Ok, you will have to experiment a bit on your own, I should get it covered with a integration test against a mssql container in the future..

**Comment by krismorte** [comment]
#29

**Comment by kagkarlsson** [comment]
Closing as it was resolved in a PR.

**Comment by jelly1400** [comment]
Hi, do you have any complete MySQL example ?

**Comment by jelly1400** [comment]
I am unable to import, please fix the repository.
import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import com.zaxxer.hikari.util.DriverDataSource;

**Comment by tomaszglinski** [comment]
@jelly1400 you know there is a difference between MSSQL and MySQL? :)

**Comment by kagkarlsson** [comment]
@jelly1400 MySQL schema is located here: https://github.com/kagkarlsson/db-scheduler/blob/master/src/test/resources/mysql_tables.sql

Could you open a separate issue for the problems you are having?

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v15.2.0` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v15.2.0))
